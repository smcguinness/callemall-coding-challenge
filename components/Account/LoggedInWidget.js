import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Chip, Tooltip } from '@material-ui/core';
import PowerIcon from '@material-ui/icons/PowerSettingsNew';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  chip: {
    margin: theme.spacing.unit,
  },
});

const LoggedInWidget = ({ classes, user, handleLogout }) => {
  const { name, photo: { thumb_link: thumbLink } } = user; // eslint-disable-line camelcase
  return (
    <Chip
      avatar={<Avatar src={thumbLink} />}
      variant="outlined"
      label={name}
      className={classes.chip}
      onDelete={handleLogout}
      deleteIcon={<Tooltip title="Logout"><PowerIcon /></Tooltip>}
    />
  );
};

LoggedInWidget.defaultProps = {
  user: {},
};

LoggedInWidget.propTypes = {
  classes: PropTypes.object.isRequired,
  handleLogout: PropTypes.func.isRequired,
  user: PropTypes.object,
};

export default withStyles(styles)(LoggedInWidget);
