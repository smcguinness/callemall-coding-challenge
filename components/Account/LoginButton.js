import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';

import MeetupIcon from '../Assets/MeetupLogo';

const styles = theme => ({
  buttonIcon: {
    marginRight: theme.spacing.unit * 2,
  },
});

class LoginButton extends Component {
  onLoginClick = () => {
    window.location = `https://secure.meetup.com/oauth2/authorize?client_id=${process.env.MEETUP_APP_KEY}&response_type=token&redirect_uri=${process.env.APPLICATION_URL}login`;
  }

  render() {
    const { classes } = this.props;
    return (
      <Button variant="outlined" onClick={this.onLoginClick}>
        <MeetupIcon className={classes.buttonIcon} fill="#f93d5d" height="24" width="24" />
        Login via Meetup
      </Button>
    );
  }
}

LoginButton.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(LoginButton);
