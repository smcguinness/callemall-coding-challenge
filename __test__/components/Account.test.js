import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';

import LoginButton from '../../components/Account/LoginButton';
import LoggedInWidget from '../../components/Account/LoggedInWidget';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Login Button', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<LoginButton />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    wrapper.setProps({ onClick: onButtonClick });
    wrapper.simulate('click');
    expect(onButtonClick.callCount).toEqual(1);
  });
});

describe('LoggedInWidget', () => {
  let wrapper;
  let user;
  beforeAll(() => {
    user = {
      name: 'Keyshawn Larson',
      photo: {
        thumb_link: 'https://s3.amazonaws.com/uifaces/faces/twitter/bereto/128.jpg',
      },
    };
  });

  it('renders correctly with a user\'s name and photo', () => {
    const handleLogout = sinon.spy();
    wrapper = shallow(<LoggedInWidget handleLogout={handleLogout} user={user} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with a user\'s name and NO photo', () => {
    const handleLogout = sinon.spy();
    wrapper = shallow(<LoggedInWidget handleLogout={handleLogout} user={{ name: user.name }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
