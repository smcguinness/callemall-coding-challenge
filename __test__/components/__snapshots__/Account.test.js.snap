// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInWidget renders correctly with a user's name and NO photo 1`] = `
<LoggedInWidget
  classes={
    Object {
      "chip": "LoggedInWidget-chip-3",
      "root": "LoggedInWidget-root-2",
    }
  }
  handleLogout={[Function]}
  user={
    Object {
      "name": "Keyshawn Larson",
    }
  }
/>
`;

exports[`LoggedInWidget renders correctly with a user's name and photo 1`] = `
<LoggedInWidget
  classes={
    Object {
      "chip": "LoggedInWidget-chip-3",
      "root": "LoggedInWidget-root-2",
    }
  }
  handleLogout={[Function]}
  user={
    Object {
      "name": "Keyshawn Larson",
      "photo": Object {
        "thumb_link": "https://s3.amazonaws.com/uifaces/faces/twitter/bereto/128.jpg",
      },
    }
  }
/>
`;

exports[`Login Button renders correctly 1`] = `
<LoginButton
  classes={
    Object {
      "buttonIcon": "LoginButton-buttonIcon-1",
    }
  }
/>
`;
